### DISCUSSION TITLE
player_hurt event e.weapon contains unexpected weapon

### DISCUSSION THREAD

**Discussion by thecatontheflat** [saul/demofile#257] [Questions] [answered]
**Describe the bug**
`player_hurt` event payload `e.weapon` contains `p250` while expected weapon (carried by the attacker) is `cz75a`

**Code to reproduce**
```
demoFile.gameEvents.on('weapon_fire', e => {
  const attacker = demoFile.entities.getByUserId(e.userid);

  if (attacker.steam64Id === '76561198137726876' && demoFile.currentTick === 89513) {
    console.log('___FIRE')
    console.log(demoFile.currentTick)
    console.log('--- weapon data')
    console.log(attacker.weapon.className)
    console.log('--- end weapon data')
    console.log(e)
  }
});

demoFile.gameEvents.on('player_hurt', e => {
  const attacker = demoFile.entities.getByUserId(e.attacker);
  const victim = demoFile.entities.getByUserId(e.userid);

  if (attacker.steam64Id === '76561198137726876' && demoFile.currentTick === 89513) {
    console.log('___HURT')
    console.log(demoFile.currentTick)
    console.log('--- weapon data')
    console.log(attacker.weapon.className)
    console.log('--- end weapon data')
    console.log(e)
  }
});
```

**Expected behaviour**
I guess it should contain `cz75a`

**Affected demos**
https://demos-europe-west2.faceit-cdn.net/csgo/2c540390-2a6b-40c1-91d2-d75ec7275d65.dem.gz

**Comment by thecatontheflat** [comment]
Found this thread (comment n 10). Is it just how server sends the event then?
https://forums.alliedmods.net/showthread.php?t=234542

**Comment by saul** [comment, marked as answer]
Yes, it is how the server sends events. `attacker.weapon.className` returns `weapon_cz75a` for me - is that not a suitable workaround?

**Comment by thecatontheflat** [comment]
Understood, thanks for confirming! 
The workaround is a suitable one — I was just not sure if that's how it was supposed to be.
